connector/signaltometricsconnector/config/config.go

```diff
@@ -299,6 +318,19 @@ func validateMetricInfo[K any](mi MetricInfo, parser ottl.Parser[K]) error {
 			return fmt.Errorf("failed to parse value OTTL expression for summary: %w", err)
 		}
 	}
+	if mi.Gauge != nil {
+		metricsDefinedCount++
+		if _, err := parser.ParseValueExpression(mi.Gauge.Value); err != nil {
+			return fmt.Errorf("failed to parse value OTTL expression for gauge: %w", err)
+		}
+		// if ExtractGrokPatterns is used, validate the key selector
+		if strings.Contains(mi.Gauge.Value, "ExtractGrokPatterns") {
+			// Ensure a [key] selector is present after ExtractGrokPatterns
+			if !regexp.MustCompile(`ExtractGrokPatterns\([^)]*\)\s*\[[^\]]+\]`).MatchString(mi.Gauge.Value) {
```

Comment:
nit: even if it's only executed once maybe we can consider moving the `MustCompile` part to var at package level to be more idiomatic (no performance impact since config validation is only executed once).